rdkPlugins/OOMCrash/source/OOMCrashPlugin.cpp

```diff
@@ -0,0 +1,202 @@
+/*
+* If not stated otherwise in this file or this component's LICENSE file the
+* following copyright and licenses apply:
+*
+* Copyright 2020 Sky UK
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#include "OOMCrashPlugin.h"
+/**
+ * Need to do this at the start of every plugin to make sure the correct
+ * C methods are visible to allow PluginLauncher to find the plugin
+ */
+REGISTER_RDK_PLUGIN(OOMCrashPlugin);
+
+/**
+ * @brief Constructor - called when plugin is loaded by PluginLauncher
+ *
+ * Do not change the parameters for this constructor - must match C methods
+ * created by REGISTER_RDK_PLUGIN macro
+ *
+ * Note plugin name is not case sensitive
+ */
+OOMCrashPlugin::OOMCrashPlugin(std::shared_ptr<rt_dobby_schema> &containerConfig,
+                             const std::shared_ptr<DobbyRdkPluginUtils> &utils,
+                             const std::string &rootfsPath)
+    : mName("OOMCrashPlugin"),
+      mContainerConfig(containerConfig),
+      mRootfsPath(rootfsPath),
+      mUtils(utils)
+{
+    AI_LOG_FN_ENTRY();
+
+    AI_LOG_FN_EXIT();
+}
+
+/**
+ * @brief Set the bit flags for which hooks we're going to use
+ *
+ * This plugin uses all the hooks so set all the flags
+ */
+unsigned OOMCrashPlugin::hookHints() const
+{
+    return (
+        IDobbyRdkPlugin::HintFlags::PostInstallationFlag |
+	IDobbyRdkPlugin::HintFlags::PostHaltFlag);
+}
+
+/**
+ *  * @brief Dobby Hook - run in host namespace *once* when container bundle is downloaded
+ *   */
+bool OOMCrashPlugin::postInstallation()
+{
+    if (!mContainerConfig)
+    {
+        AI_LOG_WARN("Container config is null");
+	return false;
+    }
+
+    const std::string source = "/opt/dobby_container_crashes";
```

Comment:
I think this directory should be read from the plugin configuration instead of being hard coded. You already have a "path" config param, you can use that for this purpose.